### DISCUSSION TITLE
Can you add a mirror into a Hubs room?

### DISCUSSION THREAD

**Discussion by emclaren** [mozilla/hubs#2883] [General Questions & Discussion] [answered]
Is it possible to add a reflective surface (like a mirror) into Hubs?

**Comment by j-conrad** [comment, marked as answer]
Usually when this question is asked, it's because someone wants to be able to see their own avatar.

There are a couple ways we could achieve this:

1. We could do something similar to the way the viewfinder image works on the snapshot camera. In this case, we would likely want to only render the avatar(s) and not the background in order to make it less costly on rendering.

2. We could allow switching to a 3rd-person camera. This might be the simplest/cheapest option for rendering.

Back the actual question of having a reflective surface-- Yes, you can currently create a material with low roughness (high glossiness), however, the material will only 'reflect' the skybox of the scene. It will not automatically reflect the rest of the scene or avatars as real time ray-tracing is not currently supported by most GPUs nor by the threejs renderer, except in some experimental work such as this: https://github.com/mrdoob/ray-tracing-renderer
